pkg/apis/metrics/register.go

```diff
@@ -48,5 +48,9 @@ func addKnownTypes() {
 	)
 }
 
-func (*RawNode) IsAnAPIObject() {}
-func (*RawPod) IsAnAPIObject()  {}
+func (obj *RawNode) SetGroupVersionKind(gvk *unversioned.GroupVersionKind) {
+	unversioned.UpdateTypeMeta(&obj.TypeMeta, gvk)
+}
+func (obj *RawPod) SetGroupVersionKind(gvk *unversioned.GroupVersionKind) {
```

Comment:
Provided by TypeMeta due to embedding.

On Thu, Dec 10, 2015 at 7:56 AM, David Eads <notifications@github.com>
wrote:

> In pkg/apis/metrics/register.go
> <https://github.com/kubernetes/kubernetes/pull/18473#discussion_r47222394>
> :
>
> > @@ -48,5 +48,9 @@ func addKnownTypes() {
> >  	)
> >  }
> >
> > -func (*RawNode) IsAnAPIObject() {}
> > -func (*RawPod) IsAnAPIObject()  {}
> > +func (obj *RawNode) SetGroupVersionKind(gvk *unversioned.GroupVersionKind) {
> > +	unversioned.UpdateTypeMeta(&obj.TypeMeta, gvk)
> > +}
> > +func (obj *RawPod) SetGroupVersionKind(gvk *unversioned.GroupVersionKind) {
>
> Corresponding GroupVersionKind()?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18473/files#r47222394>.
>